### PR TITLE
Feature: Add Creature:GetCreatureByEntry(entry) as well as Global GetCreatureTemplate(entry)

### PR DESCRIPTION
**Pull request by New-HavenWotLK**
Integrated 2 new methods to gain creature entries =)

### CODE REVIEW

**Comment by New-HavenWotLK** [comment]
@Helias / @r-o-b-o-t-o: got one more... i am on fire! =D

**Comment by 55Honey** [comment]
Thanks for the PR. I'm not sure if the naming of the new methods is consistent with the core. Interested to hear what others think.

**Comment by Helias** [comment]
> Thanks for the PR. I'm not sure if the naming of the new methods is consistent with the core. Interested to hear what others think.

Are you referring to `GetCreatureById(entry)`? we could call it `GetCreatureTemplate` to be more coherent with the core, what do you think?

**Comment by New-HavenWotLK** [comment]
> > Thanks for the PR. I'm not sure if the naming of the new methods is consistent with the core. Interested to hear what others think.
> 
> Are you referring to `GetCreatureById(entry)`? we could call it `GetCreatureTemplate` to be more coherent with the core, what do you think?

can change the global method according to that =)
would have no problem with it ^^

**Comment by New-HavenWotLK** [comment]
sooo... changed now the global method to GetCreatureTemplate(entry) =)

File: src/LuaEngine/CreatureMethods.h [hunk 1]
```diff
@@ -1401,6 +1401,18 @@ auto const& threatlist = creature->GetThreatMgr().GetThreatList();
 #endif
         return 1;
     }
+    
+    /**
+    * Retreives the entry ID of the [Creature].
+    *
+    * @param uint32 entry : the Creature ID to find.
+    */
```
Comment (r-o-b-o-t-o) [inline src/LuaEngine/CreatureMethods.h:1409]:
```suggestion

    /**
     * Retreives the entry ID of the [Creature].
     *
     * @param uint32 entry : the Creature ID to find.
     */
```

File: src/LuaEngine/GlobalMethods.h [hunk 2]
```diff
@@ -265,6 +265,18 @@ namespace LuaGlobalFunctions
         Eluna::Push(L, eMapMgr->FindMap(mapid, instance));
         return 1;
     }
+    
+    /**
+    * Retreives the entry ID of the [Creature]
+    *
+    * @param uint32 entry : the Creature ID to find
+    */
```
Comment (r-o-b-o-t-o) [inline src/LuaEngine/GlobalMethods.h:273]:
```suggestion

    /**
     * Retreives the entry ID of the [Creature]
     *
     * @param uint32 entry : the Creature ID to find
     */
```

File: src/LuaEngine/LuaFunctions.cpp [hunk 3]
```diff
@@ -116,7 +116,8 @@ luaL_Reg GlobalMethods[] =
     { "PrintError", &LuaGlobalFunctions::PrintError },
     { "PrintDebug", &LuaGlobalFunctions::PrintDebug },
     { "GetActiveGameEvents", &LuaGlobalFunctions::GetActiveGameEvents },
-
+    { "GetCreatureTemplate", &LuaGlobalFunctions::GetCreatureTemplate },
+
```
Comment (r-o-b-o-t-o) [inline src/LuaEngine/LuaFunctions.cpp:120]:
```suggestion
    { "GetCreatureTemplate", &LuaGlobalFunctions::GetCreatureTemplate },

```

**Review by r-o-b-o-t-o — CHANGES_REQUESTED** [review]

**Comment by r-o-b-o-t-o** [comment]
Reading this again, I'm not sure what you're trying to achieve with `GetCreatureByEntry`? You're returning a `CreatureTemplate` which is not even registered in Eluna 🤔

**Comment by New-HavenWotLK** [comment]
> Reading this again, I'm not sure what you're trying to achieve with `GetCreatureByEntry`? You're returning a `CreatureTemplate` which is not even registered in Eluna 🤔

wanted in the end a creature method to gain directly a creature with a specifc entry so it doesn't need a for loop iteration inside a function to get a specific creature via creaturesInRange(range, entry)
The global one i made in the end so we can make an array of npc's and iterate then over this array on e.g. creaturedeath for all those creatures without the need of extra writing for every single creature a RegisterCreatureEvent line ^^

would the creature method eventually be accomplishable with this changed version here:

int GetCreatureByEntry(lua_State* L, Creature* creature)
    {
        uint32 entry = creature->GetEntry();
        CreatureTemplate const* cEntry = sObjectMgr->GetCreatureTemplate(entry);
        if (cEntry)
            Eluna::Push(L, cEntry->Entry);
        
        return 1;
    }

**Comment by hahalmtdhl** [comment]
Can you tell me your ultimate goal?  :)

**Comment by New-HavenWotLK** [comment]
> Can you tell me your ultimate goal? :)

the goal behind this is to get specific creatures at the same bossroom and to activate them after a specific time...

means... they already are spawned in but are kinda "deactivated" i want then in the end that boss gains the creature with entry xyz after e.g. 60 secs and activates them... but without the need of a for loop where it retreives the creatures out of a table from GetCreaturesInRange(range, entry)

so kinda like:

```
local check = creature:GetCreatureByEntry(NPC_EXAMPLE)
local checkFlags = check:GetUnitFlags()

if checkFlags == creatureFlags then
    check:SetUnitFlags(0)
    WaveOneOnCombat(event, creature, target)
end
```

**Comment by New-HavenWotLK** [comment]
anyways...

it might not be my best idea... ROFL... was kinda stoned when i had this idea... ahahaha

so just let me know if u should scrap the creature method since the global method  would also be more than enough ^^

since then it still is checkable via the global one and IsInRange(target, range)

**Comment by GoMateoGo** [comment]
According to your code, you should have already obtained creaturePtr

So there's no need to obtain creation through Entry again. Relying solely on Entry is not enough Entry is just an Index

All operations on Creature are performed through class pointers, not Creature ->entry



My guess:

You want to rewrite some boss AI within a replica

Are you currently obtaining CreaturePtr through GetCreaturesInRange?

But based on the code you displayed, you have already obtained it

**Comment by r-o-b-o-t-o** [comment]
> so just let me know if u want me to scrap the creature method since the global method would also be more than enough in the end ^^

Yes please

**Pull request closed without merging**